### ISSUE TITLE
Ginkgo CI: leftover cruft in workspace 

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#1907]
After running the Ginkgo build multiple times, I get a ton of copies created of `src/github.com/cilium/cilium` in the Jenkins workspace:
```
root@jenkins-slave-4:~/jenkins/workspace/cilium_cilium_PR-1867-YBM7UJNQ3AVKIBNFUOLF2PCHUBI24WUUUTDQOQ7EOQD5KPDRDLEA/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium#
```

Build failure: https://jenkins.cilium.io/job/cilium/job/cilium/job/PR-1867/21/execution/node/28/log/:
```
22:36:41 Command: "rsync" "--verbose" "--archive" "--delete" "-z" "--copy-links" "--no-owner" "--no-group" "--rsync-path" "sudo rsync" "-e" "ssh -p 2201 -o LogLevel=FATAL  -o ControlMaster=auto -o ControlPath=/tmp/ssh.421 -o ControlPersist=10m  -o IdentitiesOnly=yes -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -i '/root/jenkins/workspace/cilium_cilium_PR-1867-YBM7UJNQ3AVKIBNFUOLF2PCHUBI24WUUUTDQOQ7EOQD5KPDRDLEA/.vagrant/machines/cilium-master-build-PR-1867-21/virtualbox/private_key'" "--exclude" ".vagrant/" "--exclude" "contrib/packaging/docker/stage" "/root/jenkins/workspace/cilium_cilium_PR-1867-YBM7UJNQ3AVKIBNFUOLF2PCHUBI24WUUUTDQOQ7EOQD5KPDRDLEA/" "vagrant@127.0.0.1:/home/vagrant/go/src/github.com/cilium/cilium"
22:36:41 Error: rsync: readlink_stat("/root/jenkins/workspace/cilium_cilium_PR-1867-YBM7UJNQ3AVKIBNFUOLF2PCHUBI24WUUUTDQOQ7EOQD5KPDRDLEA/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium") failed: Too many levels of symbolic links (40)
22:36:41 rsync: write failed on "/home/vagrant/go/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/src/github.com/cilium/cilium/vendor/github.com/imdario/mergo/LICENSE": No space left on device (28)
22:36:41 rsync error: error in file IO (code 11) at receiver.c(393) [receiver=3.1.1]
```

Isolate the cause of this issue in the Ginkgo tests.

**Comment by ianvernon** [comment]
This is due to the fact that a symlink is created within the Cilium repository to itself, which messes with rsync and Vagrant. This will be fixed by #1867

**Comment by ianvernon** [comment]
Fixed by #1867
